### ISSUE TITLE
apt-get upgrade and dpkg remove issues input/output error

### ISSUE THREAD

**Issue by msleal** [microsoft/WSL#2319]
Hi,

  Trying to upgrade the system, I started to get some input/output errors from apt-get upgrade and dpkg --remove. Looking at the specific paths input/output errors the utilities were complaining, just issuing a "touch" on that directory fixed the problem (all paths giving input/output error were not present on the filesystem anymore). So, seems like an issue on the filesystem that a FSCK could fix easily. After executing a "touch" in each one of the directories with issues (and removing the test file afterwards), I could upgrade the system without any problem (the upgrade was not working not because of the installation of new software, but becuase of the removing of old ones).
  There is a way to run FSCK on the WSL? Or is it executed at startup or something?
  Thanks! 

Microsoft Windows [Version 10.0.15063]

I was executing "sudo apt-get upgrade" and the error was input/output error for random paths (one for each package needed to be removed).

The expected behavior was the system remove the package without any issues.

**Comment by sunilmut** [comment]
@msleal - Thanks for your post. Can you please share the exact repro steps? I would like to give it a try.

**Comment by msleal** [comment]
Hi @sunilmut I have just tried to upgrade the system:
sudo apt-get update
sudo apt-get upgrade
Thanks!

**Comment by msleal** [comment]
Hi @sunilmut Just tried now to update and see the exact error:
$ sudo apt-get upgrade
Lendo listas de pacotes... Pronto
Construindo árvore de dependências
Lendo informação de estado... Pronto
Calculando atualização... Pronto
Os pacotes a seguir serão mantidos em suas versões atuais:
  libmirclient9 python3-software-properties software-properties-common software-properties-gtk
Os pacotes a seguir serão atualizados:
  libasn1-8-heimdal libepoxy0 libgssapi3-heimdal libhcrypto4-heimdal libheimbase1-heimdal libheimntlm0-heimdal libhx509-5-heimdal libkrb5-26-heimdal libroken18-heimdal libwind0-heimdal
10 pacotes atualizados, 0 pacotes novos instalados, 0 a serem removidos e 4 não atualizados.
É preciso baixar 975 kB de arquivos.
Depois desta operação, 0 B adicionais de espaço em disco serão usados.
Você quer continuar? [S/n]
Obter:1 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libroken18-heimdal amd64 1.7~git20150920+dfsg-4ubuntu1.16.04.1 [41,4 kB]
Obter:2 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libasn1-8-heimdal amd64 1.7~git20150920+dfsg-4ubuntu1.16.04.1 [174 kB]
Obter:3 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libhcrypto4-heimdal amd64 1.7~git20150920+dfsg-4ubuntu1.16.04.1 [85,0 kB]
Obter:4 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libheimbase1-heimdal amd64 1.7~git20150920+dfsg-4ubuntu1.16.04.1 [29,3 kB]
Obter:5 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libwind0-heimdal amd64 1.7~git20150920+dfsg-4ubuntu1.16.04.1 [47,8 kB]
Obter:6 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libhx509-5-heimdal amd64 1.7~git20150920+dfsg-4ubuntu1.16.04.1 [107 kB]
Obter:7 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libkrb5-26-heimdal amd64 1.7~git20150920+dfsg-4ubuntu1.16.04.1 [202 kB]
Obter:8 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libheimntlm0-heimdal amd64 1.7~git20150920+dfsg-4ubuntu1.16.04.1 [15,1 kB]
Obter:9 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libgssapi3-heimdal amd64 1.7~git20150920+dfsg-4ubuntu1.16.04.1 [96,1 kB]
Obter:10 http://archive.ubuntu.com/ubuntu xenial-updates/main amd64 libepoxy0 amd64 1.3.1-1ubuntu0.16.04.2 [177 kB]
Baixados 975 kB em 29s (33,2 kB/s)
(Lendo banco de dados ... 56294 ficheiros e directórios actualmente instalados.)
A preparar para desempacotar .../libroken18-heimdal_1.7~git20150920+dfsg-4ubuntu1.16.04.1_amd64.deb ...
A descompactar libroken18-heimdal:amd64 (1.7~git20150920+dfsg-4ubuntu1.16.04.1) sobre (1.7~git20150920+dfsg-4ubuntu1) ...
A preparar para desempacotar .../libasn1-8-heimdal_1.7~git20150920+dfsg-4ubuntu1.16.04.1_amd64.deb ...
A descompactar libasn1-8-heimdal:amd64 (1.7~git20150920+dfsg-4ubuntu1.16.04.1) sobre (1.7~git20150920+dfsg-4ubuntu1) ...
A preparar para desempacotar .../libhcrypto4-heimdal_1.7~git20150920+dfsg-4ubuntu1.16.04.1_amd64.deb ...
A descompactar libhcrypto4-heimdal:amd64 (1.7~git20150920+dfsg-4ubuntu1.16.04.1) sobre (1.7~git20150920+dfsg-4ubuntu1) ...
A preparar para desempacotar .../libheimbase1-heimdal_1.7~git20150920+dfsg-4ubuntu1.16.04.1_amd64.deb ...
A descompactar libheimbase1-heimdal:amd64 (1.7~git20150920+dfsg-4ubuntu1.16.04.1) sobre (1.7~git20150920+dfsg-4ubuntu1) ...
A preparar para desempacotar .../libwind0-heimdal_1.7~git20150920+dfsg-4ubuntu1.16.04.1_amd64.deb ...
A descompactar libwind0-heimdal:amd64 (1.7~git20150920+dfsg-4ubuntu1.16.04.1) sobre (1.7~git20150920+dfsg-4ubuntu1) ...
A preparar para desempacotar .../libhx509-5-heimdal_1.7~git20150920+dfsg-4ubuntu1.16.04.1_amd64.deb ...
A descompactar libhx509-5-heimdal:amd64 (1.7~git20150920+dfsg-4ubuntu1.16.04.1) sobre (1.7~git20150920+dfsg-4ubuntu1) ...
A preparar para desempacotar .../libkrb5-26-heimdal_1.7~git20150920+dfsg-4ubuntu1.16.04.1_amd64.deb ...
A descompactar libkrb5-26-heimdal:amd64 (1.7~git20150920+dfsg-4ubuntu1.16.04.1) sobre (1.7~git20150920+dfsg-4ubuntu1) ...
A preparar para desempacotar .../libheimntlm0-heimdal_1.7~git20150920+dfsg-4ubuntu1.16.04.1_amd64.deb ...
A descompactar libheimntlm0-heimdal:amd64 (1.7~git20150920+dfsg-4ubuntu1.16.04.1) sobre (1.7~git20150920+dfsg-4ubuntu1) ...
A preparar para desempacotar .../libgssapi3-heimdal_1.7~git20150920+dfsg-4ubuntu1.16.04.1_amd64.deb ...
A descompactar libgssapi3-heimdal:amd64 (1.7~git20150920+dfsg-4ubuntu1.16.04.1) sobre (1.7~git20150920+dfsg-4ubuntu1) ...
A preparar para desempacotar .../libepoxy0_1.3.1-1ubuntu0.16.04.2_amd64.deb ...
A descompactar libepoxy0:amd64 (1.3.1-1ubuntu0.16.04.2) sobre (1.3.1-1ubuntu0.16.04.1) ...
dpkg: erro ao processar o arquivo /var/cache/apt/archives/libepoxy0_1.3.1-1ubuntu0.16.04.2_amd64.deb (--unpack):
 impossível executar "stat" de './usr/share/doc/libepoxy0' (que eu estava prestes a instalar): Input/output error
A processar 'triggers' para libc-bin (2.23-0ubuntu9) ...
Erros foram encontrados durante o processamento de:
 /var/cache/apt/archives/libepoxy0_1.3.1-1ubuntu0.16.04.2_amd64.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)

It really seems an inconsistent filesystem, but I'm not able to run an FSCK...
Thank you!

**Comment by msleal** [comment]
Hi,

Some more info..
After the upgrade of my Windows 10, I also did the upgrade of the Ubuntu system. Seems like the date/time configuration of my machine was lost in the upgrade process, and after the upgrade it was configured for UTC. I just looked into it because of the error I was getting while trying to create a regular file in the "/usr/" directory:

root@machine:/# cd /usr/lib/x86_64-linux-gnu/
root@machine:/usr/lib/x86_64-linux-gnu# ls -l pkgconfig
ls: cannot access 'pkgconfig': No such file or directory
root@machine:/usr/lib/x86_64-linux-gnu# touch pkgconfig
touch: setting times of 'pkgconfig': No such file or directory

All the files that need to be upgraded, gives the same error (input/output error) in the apt-get command. If I try to create crazy files (e.g.: xfsdfsdfssd), it works. But filenames that are involved with packages (that need to be upgraded), gives error for deleting or creating (what makes me think it is really filesystem consistency). 

I do not know how to execute a FSCK on the Linux subsystem inside Windows, but I think something along these lines could fix the issue.

 thanks!
